core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -25,161 +25,177 @@
           + " SET "
-          + TABLE_NAMESPACE
-          + " = ? , "
-          + TABLE_NAME
-          + " = ? "
+          + TABLE_VIEW_NAMESPACE
+          + " = ?, "
+          + TABLE_VIEW_NAME
+          + " = ?"
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
           + " = ? AND "
-          + TABLE_NAME
-          + " = ? ";
-  static final String DROP_TABLE_SQL =
+          + TABLE_VIEW_NAME
+          + " = ? AND "
+          + TABLE_VIEW_FLAG
+          + " = ?";
+  static final String DROP_SQL =
       "DELETE FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
+          + "  = ? AND "
+          + TABLE_VIEW_NAME
           + " = ? AND "
-          + TABLE_NAME
-          + " = ? ";
-  static final String GET_NAMESPACE_SQL =
+          + TABLE_VIEW_FLAG
+          + " = ?";
+  private static final String GET_NAMESPACE_SQL =
       "SELECT "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
           + " FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
-          + " ( "
-          + TABLE_NAMESPACE
+          + " ("
+          + TABLE_VIEW_NAMESPACE
           + " = ? OR "
-          + TABLE_NAMESPACE
-          + " LIKE ? ESCAPE '\\' "
-          + " ) "
+          + TABLE_VIEW_NAMESPACE
+          + " LIKE ? ESCAPE '\\')"
           + " LIMIT 1";
   static final String LIST_NAMESPACES_SQL =
       "SELECT DISTINCT "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
           + " FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ? AND "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
           + " LIKE ?";
-  static final String LIST_ALL_TABLE_NAMESPACES_SQL =
+  static final String LIST_ALL_NAMESPACES_SQL =
       "SELECT DISTINCT "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
           + " FROM "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " WHERE "
           + CATALOG_NAME
           + " = ?";
-  static final String DO_COMMIT_CREATE_TABLE_SQL =
+  private static final String DO_COMMIT_CREATE_SQL =
       "INSERT INTO "
-          + CATALOG_TABLE_NAME
+          + CATALOG_TABLE_VIEW_NAME
           + " ("
           + CATALOG_NAME
           + ", "
-          + TABLE_NAMESPACE
+          + TABLE_VIEW_NAMESPACE
           + ", "
-          + TABLE_NAME
+          + TABLE_VIEW_NAME
           + ", "
           + JdbcTableOperations.METADATA_LOCATION_PROP
           + ", "
           + JdbcTableOperations.PREVIOUS_METADATA_LOCATION_PROP
+          + ", "
+          + TABLE_VIEW_FLAG
```

Comment:
I think it breaks existing Iceberg table queries in JDBC table as schema is changed?? 

we do need some way to handle without breaking compatibility. Manually asking users to upgrade and migrate may be a pain.

Reply:
Automatically updating the schema before any SQL query seems like a right thing to do. But might slightly degrade performance as it will be two calls now.

Reply:
What I plan to do is to update the schema (if needed) in the `JdbcCatalog#initialize()` method. As it's where we create the SQL tables, I think it makes sense to update the SQL tables if needed.